pkg/scheduler/scheduler.go

```diff
@@ -221,7 +293,20 @@ func (s *ipClaimScheduler) processOldService(svc *v1.Service) {
 	}
 	for _, ip := range svc.Spec.ExternalIPs {
 		if _, ok := refs[ip]; !ok {
-			err := deleteIPClaim(s.ExtensionsClientset, ip, s.DefaultMask)
+			pool, err := getPoolForAllocated(s.ExtensionsClientset, ip)
+			if err != nil {
+				glog.Errorf("Unable to determine whether ip %v was auto-allocated. Detals: %v", ip, err)
+			}
+			//if IP was auto allocated, use mask from its pool
+			if pool != nil {
+				err = deleteIPClaim(
+					s.ExtensionsClientset, ip, strings.Split(pool.Spec.CIDR, "/")[1])
+
+				defer removeAllocation(s.ExtensionsClientset, pool, ip)
```

Comment:
what the point of defer? please ensure that only single update to pool is done for whole loop